src/parse.rs

```diff
@@ -299,6 +299,36 @@ pub fn parse_ids(
     }
 }
 
+pub fn parse_idle(
+    mut lines: &[u8],
+    unsolicited: &mut mpsc::Sender<UnsolicitedResponse>,
+) -> Result<()> {
+    while !lines.is_empty() {
+        match imap_proto::parse_response(lines) {
+            Ok((rest, data)) => {
+                lines = rest;
+                if let Some(resp) = handle_unilateral(data, unsolicited) {
```

Comment:
Are unilateral responses the only thing that can come from IDLE?

Reply:
IDLE can only send untagged responses, but there can be some that are not currently handled by the UnsolicitedResponse enum, like FETCH responses.

Reply:
We probably want to handle these here too then, though in that case I wonder if we need a different channel since the types we send would be different?

Reply:
I think a Fetch item should just be added to the UnsolicitedResponse enum.

Reply:
Maybe, although that also seems weird if you can't generally receive unsolicitied `FETCH` response (are there other responses IDLE adds too?).

Reply:
You can, see 7.4.2 of rfc 350: This response occurs as the result of a FETCH or
      STORE command, as well as by unilateral server decision (e.g.,
      flag updates).
There can also be OK/NO/BAD responses. E.g. Dovecot sends 'OK Still here' periodically, I currently filter OK out in parse_idle, but it could be returned to the user too. They may be sent to other commands too.

Reply:
That sounds to me like we should really make the IDLE channel totally separate, and have it send only the `FETCH` and `STORE` responses. Unilateral responses would still go to the original unilateral channel. Maybe we make them both `crossbeam-channel` channels so that the user can select over both of them at the same time?

Reply:
I don't think so. A FETCH response can really occur at any time just like an EXISTS, EXPUNGE or RECENT response. You'll get one with any command sent after a message has been marked as read for instance. You really want all these responses when you use idle(). Besides, if you use separate channels, you lose the chronology of those responses and thus can't maintain the id<->message association.

Reply:
Sounds good to me! I guess we then need to update `handle_unilateral` to also handle those responses :)

Reply:
Right, I'll give it a try tomorrow.